RecyclerViewKotlin/app/src/main/java/com/example/recyclersample/flowerList/FlowersListViewModel.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.recyclersample.flowerList
+
+import android.content.Context
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.recyclersample.data.DataSource
+import com.example.recyclersample.data.Flower
+import kotlin.random.Random
+
+class FlowersListViewModel(val dataSource: DataSource) : ViewModel() {
+
+    val flowersLiveData = dataSource.getFlowerList()
+
+    /* If the name and description are present, create new Flower and add it to the datasource */
+    fun insertFlower(flowerName: String?, flowerDescription: String?) {
+        if(flowerName == null || flowerDescription == null){
```

Comment:
Missing space after `if`, which tells me that ktlint needs to be run again.

Reply:
Bump

Reply:
Hmmm, I checked again today and it didn't flag this